### DISCUSSION TITLE
[Art competition] Search space pruning

### DISCUSSION THREAD

**Discussion by KunJeong** [prosyslab-classroom/cs424-program-reasoning#254] [Art competition]
![DALL·E 2022-11-30 16 11 31 - thanos snapping his fingers while looking up at a starry night sky, cartoon style](https://user-images.githubusercontent.com/10477004/204730936-a30b94e2-cf08-42fa-8acf-0acbcd4c0853.png)

## Short Description
For me, [search space pruning] means [thanos snapping his fingers while looking up at a starry night sky, cartoon style]".

## Long Description
In a galaxy full of possible candidate programs, it is important to prune those candidates quickly and efficiently. With the snap of his fingers, thanos prunes away half of the universe.. with ten more snaps, we will be able to find a complex program in no time.

**Comment by KihongHeo** [comment]
What are the optimization criteria of Thanos? Equivalence reduction? or what?
